drgn_tools/lsmod.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+import argparse
+
+from drgn import Object
+from drgn import Program
+from drgn.helpers.common.memory import identify_address
+from drgn.helpers.linux.list import list_for_each_entry
+
+from drgn_tools.corelens import CorelensModule
+
+
+def get_parm_valf(prog: Program, parm_val: str, kp: Object) -> int:
+    """
+    converting endian to big endian
+
+    :param prog: drgn program
+    :param parm_val: kernel_param.arg is the paramter value that is converted to big endian
+    :param kp: kernel_param
+    :returns: int after converting from hex
+    """
+    parm_val1 = ""
+    for i in range(0, len(parm_val), 2):
+        parm_val1 = (
+            str(prog.read(kp.arg.value_(), 2).hex())[i : i + 2] + parm_val1
+        )
+    return int(str(parm_val1), 16)
+
+
```

Comment:
I'm not really sure I understand the purpose of this function, but I don't think we need it - see my approach for reading integers.

In general, even if you didn't use the approach of casting with drgn, you might find the following functions from drgn helpful:

Program.read_u8
Program.read_u16
Program.read_u32
Program.read_u64